src/debug/shared/dbgtransportsession.cpp

```diff
@@ -57,6 +57,11 @@ HRESULT DbgTransportSession::Init(DebuggerIPCControlBlock *pDCB, AppDomainEnumer
     // cleanup necessary.
     memset(this, 0, sizeof(*this));
 
+    // Because of the above memset the embeded classes/structs need to be reinitialized especially
+    // the two way pipe; it expects the in/out handles to be -1 instead of 0.
```

Comment:
It seems a bit strange to have the constructor run, then zero them out, and then run the constructors again, but I don't know that there's anything technically wrong with it.  Presumably no one could be using the lock or pipe as they were previously initialized before this point.